AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/SQLStatement+Select.swift

```diff
@@ -9,78 +9,49 @@ import Amplify
 import Foundation
 import SQLite
 
-/// Represents a `select` SQL statement associated with a `Model` instance and
-/// optionally composed by a `ConditionStatement`.
-struct SelectStatement: SQLStatement {
-
-    let modelType: Model.Type
-    let conditionStatement: ConditionStatement?
-    let paginationInput: QueryPaginationInput?
-
-    // TODO remove this once sorting support is added to DataStore
-    // Used by plugin to order and limit results for system table queries
-    let additionalStatements: String?
-    let namespace = "root"
-
-    init(from modelType: Model.Type,
-         predicate: QueryPredicate? = nil,
-         paginationInput: QueryPaginationInput? = nil,
-         additionalStatements: String? = nil) {
-        self.modelType = modelType
-
-        var conditionStatement: ConditionStatement?
-        if let predicate = predicate {
-            let statement = ConditionStatement(modelType: modelType,
-                                               predicate: predicate,
-                                               namespace: namespace[...])
-            conditionStatement = statement
-        }
-        self.conditionStatement = conditionStatement
-        self.paginationInput = paginationInput
-        self.additionalStatements = additionalStatements
-    }
-
-    var stringValue: String {
+/// Support data structure used to hold information about `SelectStatement` than
+/// can be used later to parse the results.
+struct SelectStatementMetadata {
+
+    typealias ColumnMapping = [String: (ModelSchema, ModelField)]
+
+    let statement: String
+    let columnMapping: ColumnMapping
+    let bindings: [Binding?]
+
+    // TODO remove additionalStatements once sorting support is added to DataStore
+    static func get(from modelType: Model.Type,
```

Comment:
sure thing, suggestions?

Reply:
can this just be the constructor for `SelectStatementMetadata`?

Reply:
make `metaData(from:)`, `statementMetaData(from:)` or why not an init method?

Reply:
> can this just be the constructor for `SelectStatementMetadata`?

using the constructor is tricky because I have to execute the entire logic inside the constructor (no other functions, etc) because I need to initialize any struct property.

That's why I went with the static factory function, otherwise the constructor becomes a mess.

Reply:
I was also thinking `.selectStatementMetadata(from:)`

Reply:
it's an internal API, so I renamed to `.metadata()`, feel free to change that later as it has no impact on customers.